CONTRIBUTING.md

```diff
@@ -255,6 +255,49 @@ pub use {
 }
 ```
 
+#### Return `Self`
+
+Use `Self` as the return type instead of naming the type.
+
+```rust
+/// A counter that is always smaller than 100.
+pub struct Counter(u32);
+
+impl Counter {
+    /// Constructs a new `Counter`.
+    pub fn new() -> Self { Self(0) }
+
+    /// Returns a counter if it is possible to create one from x.
+    pub fn maybe(x: u32) -> Option<Self> {
+        match x {
+            x if x >= 100 => None,
+            c => Some(Counter(c)),
+        }
+    }
+}
+
+impl TryFrom<u32> for Counter {
+    type Error = TooBigError;
+
+    fn try_from(x: u32) -> Result<Self, Self::Error> {
+        if x >= 100 {
+            return Err(TooBigError);
+        }
+        Ok(Counter(x))
+    }
+}
+```
+
+When constructing the return value use `Self` or the actual type, as you prefer.
+
+```rust
+impl From<some::Error> for Error {
+    fn from(e: some::Error) -> Self { Self::Some(e) }
+}
+```
+
+When constructing the return value for errors use `Self`.
```

Comment:
You mean `Self::Error`?

Reply:
I meant to mean `Self::Foo` where `Foo` is the variant instead of `ReallyLongDescriptiveError::Foo`

Reply:
Ah, you mean enums. I think writing enums would be better since not all errors are enums. :)

Reply:
Ah yes, good thinking. Thanks